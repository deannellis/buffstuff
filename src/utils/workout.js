import moment from 'moment';

export const getWorkouts = ({ a = 0, b = 0 }) => {
    let workouts = []
    if(a == 0) {
        workouts.push('overhead', 'chinup')
    } else {
        workouts.push('bench', 'row')
    }
    if(b == 0 || b == 2) {
        workouts.push('squat');
    } else {
        workouts.push('deadlift');
    }
    return workouts;
}

export const getDisplayName = name => {
    switch(name) {
        case 'bench':
            return 'Bench Press';
        case 'row':
            return 'Row';
        case 'squat':
            return 'Squat';
        case 'deadlift':
            return 'Deadlift';
        case 'overhead':
            return 'Overhead Press';
        case 'chinup':
            return 'Chin-Ups';
        default:
            return name;
    }
};

export const getWarmupWeights = workWeight => {
    const warmup25 = workWeight * 0.25 > 44 ? workWeight * 0.25 : 44;
    const warmup50 = workWeight * 0.50 > 44 ? workWeight * 0.50 : 44;
    const warmup75 = workWeight * 0.75 > 44 ? workWeight * 0.75 : 44;
    return [ warmup25, warmup50, warmup75 ];
}

export const getWeightDistribution = weight => {
    if(weight === 44) {
        return [ 0, 44, 0]
    }
    const oneSide = (weight - 44) / 2;
    return [ oneSide, 44, oneSide ];
}

export const getMonthWorkouts = workouts => {
    let dates = {};
    workouts.forEach(workout => {
        let month = moment(workout.created).format("MMMM");
        let day = parseInt(moment(workout.created).format("D"));
        if(dates.hasOwnProperty(month)) {
            dates[month].push(day);
        } else {
            dates[month] = [ day ];
        }
    });
    // console.log(dates);
    return dates;
}

export const getWorkoutDays = (start, daysInMonth) => {
    let workoutDays = [];
    let checkEven = false;
    let weekday = start + 1;
    for(let d = 1; d <= daysInMonth; d++) {
        if(checkEven) {
            if(d%2 == 0) workoutDays.push(d)
        } else {
            if(d%2 !== 0) workoutDays.push(d)
        }
        if(weekday == 7){
            weekday = 0;
            checkEven = !checkEven;
        };
        weekday ++;
    }
    return workoutDays;
};
