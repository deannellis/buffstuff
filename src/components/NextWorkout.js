import React from 'react';
import { getWorkouts, getDisplayName } from '../utils/workout';
import Button from './Button';

const NextWorkout = (props) => {
    console.log('var', getWorkouts(props.liftVariant));
    return (  
        <div className="next-workout dashboard__card">
            <h2>Next Workout</h2>
            <div className="next-workout__lifts">
                {getWorkouts(props.liftVariant).map(workout => {
                    return (
                        <p key={workout} className="next-workout__lift">
                            {getDisplayName(workout)}
                            <span className="next-workout__weight">
                                {workout !== 'chinup' ? ` @${props.masterWeights[workout]}lbs` : ''}
                            </span>
                        </p>
                    );
                })}
            </div>
            <Button variant="primary" className="next-workout__button">Start workout</Button>
        </div>
    );
}
 
export default NextWorkout;