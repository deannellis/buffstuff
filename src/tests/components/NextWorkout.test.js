import React from 'react';
import { shallow } from 'enzyme';

import NextWorkout from '../../components/NextWorkout';
import { weights } from '../fixtures/workout';

test('should render NextWorkout without props', () => {
    const wrapper = shallow(<NextWorkout />);
    expect(wrapper).toMatchSnapshot();
});

test('should render NextWorkout with props', () => {
    const wrapper = shallow(<NextWorkout 
        liftVariant={{ a: 1, b: 1 }} 
        masterWeights={weights}
    />);
    expect(wrapper).toMatchSnapshot();
});