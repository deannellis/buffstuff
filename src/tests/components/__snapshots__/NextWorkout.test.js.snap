// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render NextWorkout with props 1`] = `
<div
  className="next-workout card"
>
  <h2>
    Next Workout
  </h2>
  <div
    className="next-workout__lifts"
  >
    <p
      className="next-workout__lift"
      key="bench"
    >
      Bench Press
      <span
        className="next-workout__weight"
      >
         @85lbs
      </span>
    </p>
    <p
      className="next-workout__lift"
      key="row"
    >
      Row
      <span
        className="next-workout__weight"
      >
         @115lbs
      </span>
    </p>
    <p
      className="next-workout__lift"
      key="deadlift"
    >
      Deadlift
      <span
        className="next-workout__weight"
      >
         @106lbs
      </span>
    </p>
  </div>
  <Button
    className="next-workout__button"
    clickHandler={[Function]}
    disabled={false}
    onClick={[Function]}
    type="button"
    variant="primary"
  >
    Start workout
  </Button>
</div>
`;

exports[`should render NextWorkout without props 1`] = `
<div
  className="next-workout card"
>
  <h2>
    Next Workout
  </h2>
  <div
    className="next-workout__lifts"
  >
    <p
      className="next-workout__lift"
      key="overhead"
    >
      Overhead Press
      <span
        className="next-workout__weight"
      >
         @0lbs
      </span>
    </p>
    <p
      className="next-workout__lift"
      key="chinup"
    >
      Chin-Ups
      <span
        className="next-workout__weight"
      />
    </p>
    <p
      className="next-workout__lift"
      key="squat"
    >
      Squat
      <span
        className="next-workout__weight"
      >
         @0lbs
      </span>
    </p>
  </div>
  <Button
    className="next-workout__button"
    clickHandler={[Function]}
    disabled={false}
    onClick={[Function]}
    type="button"
    variant="primary"
  >
    Start workout
  </Button>
</div>
`;
